mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -1014,13 +1014,14 @@ OpFoldResult arith::MaxNumFOp::fold(FoldAdaptor adaptor) {
   if (getLhs() == getRhs())
     return getRhs();
 
-  // maxnumf(x, -inf) -> x
-  if (matchPattern(adaptor.getRhs(), m_NegInfFloat()))
+  // maxnumf(x, NaN) -> x
+  if (matchPattern(adaptor.getRhs(), m_NaNFloat()))
     return getLhs();
 
-  return constFoldBinaryOp<FloatAttr>(
-      adaptor.getOperands(),
-      [](const APFloat &a, const APFloat &b) { return llvm::maximum(a, b); });
+  return constFoldBinaryOp<FloatAttr>(adaptor.getOperands(),
+                                      [](const APFloat &a, const APFloat &b) {
+                                        return llvm::maximumnum(a, b);
+                                      });
```

Comment:
nit: could we pass in just `llvm::maximumnum` here instead of the lambda? I don't know if it's overloaded or not.

Reply:
Yes I think that would work. Should I update other similar lambdas in the fold implementations?

Reply:
In general, I'd only update the code that relevant to this PR, and submit any other cleanups as a separate PR